kingpin/actors/aws/ecs.py

```diff
@@ -406,7 +420,369 @@ def _execute(self):
+            family: task definition family.
+            revision: task definition revision.
+        """
+        deployment_configuration = self.service_definition[
+            'deploymentConfiguration']
+        update_parameters = dict(
+            cluster=self.option('cluster'),
+            service=service_name,
+            taskDefinition='{0}:{1}'.format(family, revision),
+            desiredCount=self.option('count'),
+            deploymentConfiguration=deployment_configuration)
+
+        self.log.info('Updating service.')
+
+        yield self.thread(
+            self.ecs_conn.update_service,
+            **update_parameters)
+
+    @gen.coroutine
+    def _register_service(self, service_name, family, revision):
+        """Registers a service.
+
+        This handles the logic of either:
+            1) Creating a new service if it doesn't exist.
+            or
+            2) Updating the existing service.
+
+        Args:
+            service_name: service_name to use.
+            family: Task Definition family.
+            revision: Task Definition revision.
+        """
+        existing_service = yield self._describe_service(service_name)
+
+        if not existing_service or (existing_service and
+                                    existing_service['status'] == 'INACTIVE'):
+            # Generate a 32 character uuid for the client token.
+            client_token = str(uuid.uuid4())
+            yield self._create_service(service_name,
+                                       family, revision,
+                                       client_token)
+        else:
+            errors = _get_immutable_field_errors(
+                old_params=existing_service,
+                new_params=self.service_definition,
+                immutable_fields=['loadBalancers', 'role'])
+
+            if errors:
+                for error in errors:
+                    self.log.error(error)
+                raise exceptions.UnrecoverableActorFailure(
+                    'Immutable fields cannot be updated. '
+                    'A new service must be created.')
+
+            yield self._update_service(service_name,
+                                       family, revision)
+        raise gen.Return()
+
+    @gen.coroutine
+    @dry('Would wait for service to deploy')
+    def _wait_for_deployment(self, service_name):
+        """Wait for service to deploy on ECS.
+
+        If the service fails to come up,
+        this will raise an exception.
+
+        Args:
+            service_name: name of the service to wait for.
+        """
+        # Used to ensure event logs are only printed once.
+        seen_events = set()
+        while True:
+            done = yield self._service_deployed(
+                service_name,
+                seen_events)
+            if done:
+                break
+            yield gen.sleep(10)
+
+    @gen.coroutine
+    def _service_deployed(self, service_name, seen_events):
```

Comment:
Not a fan of this function name -- it sounds like its saying "the service IS deployed", not "checking IF the service is deployed". Sure, once I read it, I get it.. but its not obvious at first glance.

Rename it to `self._is_service_deployed()`?